platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java

```diff
@@ -385,9 +393,62 @@ boolean onTap(PointF tapPoint) {
       }
     }
 
-    MarkerHit markerHit = getMarkerHitFromTouchArea(tapPoint);
+    Marker marker = getMarkerBelowPoint(tapPoint);
+    return marker != null && isClickHandledForMarker(marker);
+  }
+
+  //region Drag Event Handlers
+
+  boolean onMoveBegin(PointF focalPoint) {
+    Marker marker = getMarkerBelowPoint(focalPoint);
+    if (marker != null && marker.isDraggable()) {
+      currentMarkerBeingDragged = marker;
+      if (onMarkerDragListener != null) {
+        onMarkerDragListener.onMarkerDragStart(marker);
+      }
+      return true;
+    }
+    return false;
+  }
+
+  boolean onMove(MoveDistancesObject movedObject, float distanceX, float distanceY) {
+    if (currentMarkerBeingDragged == null || !currentMarkerBeingDragged.isDraggable()) {
+      return false;
+    }
+    PointF screenLocation = new PointF(movedObject.getCurrentX(), movedObject.getCurrentY());
+    screenLocation.offset(-distanceX, -distanceY);
```

Comment:
The current coordinates of the `MoveDistancesObject` reflect the most recent position, no need for offsetting them further.